library/nameops.ml

```diff
@@ -124,34 +124,86 @@ let atompart_of_id id = fst (repr_ident id)
 
 (* Names *)
 
-let out_name = function
-  | Name id -> id
-  | Anonymous -> failwith "Nameops.out_name"
+module type ExtName =
+sig
 
-let name_fold f na a =
+  include module type of struct include Names.Name end
+
+  val fold_left : ('a -> Id.t -> 'a) -> 'a -> t -> 'a
+  val fold_right : (Id.t -> 'a -> 'a) -> t -> 'a -> 'a
```

Comment:
Could we keep only one `fold` function? In fact, I don't even understand why we would like to have such functions; I would expect to find a function with type `(Id.t -> 'a) -> 'a -> t -> 'a`, but there is none.